### ISSUE TITLE
Fn-toggle stops working after some time

### ISSUE THREAD

**Issue by dieterdemeyer** [bikkelbroeders/TouchBarDemoApp#46]
I'm afraid this issue has returned.

I'm using the latest version of TouchBarServer and I've noticed that the Fn key stops working after some time. It takes a lot longer now before it stops working (days even)..

**Comment by sonvirgo** [comment]
me too

**Comment by MekVox** [comment]
me too

**Comment by sonvirgo** [comment]
found this, always work https://red-sweater.com/touche/

**Comment by MekVox** [comment]
Thanks 😊

**Comment by robbertkl** [comment]
Could you check again with the latest version? Not much has changed for how the modifier presses are handled, but it does allow changing the toggle key.

Perhaps you could use control as a modifier key and see if it stops working after a while as well?

Also, I'm curious how to "make it work again". Does it require quitting and restarting TouchBarServer, or can you also disable and then re-enable the on-screen Touch Bar using the app's status menu?

**Comment by MekVox** [comment]
Hello,

are you one of the developer?

I’ve download the 1.4 version two days ago.
It works fine, thanks!
1. I still can’t change the toggle key (no setting).
2. The touch bar still is not a real dock, so all the windows opened go above it.

Before the update when the touch bar server app didn’t work, I usually quit all the process relative to “touch bar” in activity monitor, or I simply restarted the Mac.

Hope to be useful.
All the best,
Mek Vox
Video Editor

mekvox@gmail.com mailto:antonellimattia@gmail.com

ELSE TEAR PRODUCTION
www.elsetearproduction.com http://www.elsetearproduction.com/

**Comment by robbertkl** [comment]
Yes, @MekVox, I'm 1 of the developers. Let's please stay on topic and keep this issue about the Fn key not working anymore after some time, instead of other issues/feature requests. Please check the [releases page](https://github.com/bikkelbroeders/TouchBarDemoApp/releases) for changelogs of each release. As you can see, v1.4 did not add the features you're asking about, we never claimed it did. As of v1.5, you can change the toggle key (see #28).

Thank you for the additional information. Why did you quit all the Touch Bar processes in Activity Monitor? Did a restart of the TouchBarServer app not work?

**Comment by MekVox** [comment]
Ok thanks!

Yes for your last question the answer is yes 😊

**Comment by robbertkl** [comment]
Just committed a possible fix for this. Build the latest source or wait until the v1.6 release.

**Comment by MekVox** [comment]
Fantastic! Thanks again!

Are you going to improve the touch bar with the overlay function?

Thanks,
Mek Vox
Video Editor

mekvox@gmail.com mailto:antonellimattia@gmail.com

ELSE TEAR PRODUCTION
www.elsetearproduction.com http://www.elsetearproduction.com/
